rust/crd/src/lib.rs

```diff
@@ -54,36 +59,54 @@ pub enum Error {
     FailedQuantityConversion,
     #[snafu(display("failed to parse value"))]
     FailedParseToFloatConversion,
+    #[snafu(display("fragment validation failure"))]
+    FragmentValidationFailure { source: ValidationError },
 }
 
-#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize, JsonSchema)]
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize, JsonSchema)]
+#[allow(clippy::derive_partial_eq_without_eq)]
 #[serde(rename_all = "camelCase")]
 pub struct SparkApplicationStatus {
     pub phase: String,
 }
 
-#[derive(Clone, Debug, Default, Deserialize, Merge, JsonSchema, Eq, PartialEq, Serialize)]
-#[serde(rename_all = "camelCase")]
+#[derive(Clone, Debug, Default, JsonSchema, PartialEq, Fragment)]
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[fragment_attrs(
+    derive(
+        Clone,
+        Debug,
+        Default,
+        Deserialize,
+        Merge,
+        JsonSchema,
+        PartialEq,
+        Eq,
+        Serialize
+    ),
```

Comment:
I would keep `SparkStorageConfig` and `SparkStorageConfigFragment` consistent.
```suggestion
        Serialize,
        ),
    allow(clippy::derive_partial_eq_without_eq),
```
```